### ISSUE TITLE
Enhancement: Forbid class property with name `constructor`

### ISSUE THREAD

**Issue by fisker** [typescript-eslint/typescript-eslint#11584]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Currently

```ts
class A {constructor = 1 }
```

is correctly rejected.

But

```ts
class A { 'constructor' = 1 }
```

which should also invalid.

### Additional Info

https://github.com/typescript-eslint/typescript-eslint/pull/11299 is kind of related.

**Comment by fisker** [comment]
I'll try to fix it.
